lib/cache/caches/ClassCache.ts

```diff
@@ -0,0 +1,32 @@
+import { Expiration } from '../expirations/Expiration';
+import { HashService } from '../../utils/hash/hash';
+import { Storage } from '../storages/Storage';
+import { Cache } from './Cache';
+
+export class ClassCache<K = any> implements Cache<K> {
+
+  constructor(
+    private readonly storage: Storage,
+    private readonly expiration: Expiration,
+    private readonly hash: HashService,
+  ) { }
+
+  public async set<V>(key: K, value: V): Promise<void> {
+    const keyHash = this.hash.calculate(key);
+
+    await this.storage.set(keyHash, value);
+    this.expiration.add(keyHash, key => this.delete(key));
+  }
+
+  public async get<V>(key: K): Promise<V> {
+    const keyHash = this.hash.calculate(key);
+
+    this.expiration.add(keyHash, key => this.delete(key));
```

Comment:
At this point the key might not be cached any more, still you add it to the expiration. I'd suggest first to check if the key is in the storage.